var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -226,14 +225,20 @@ def cmake_args(self):
                 if not cuda_arch == "none":
                     kokkos_arch_name = self.spack_cuda_arch_map[cuda_arch]
                     spack_microarches.append(kokkos_arch_name)
+
         kokkos_microarch_name = self.spack_micro_arch_map[spec.target.name]
         if kokkos_microarch_name:
             spack_microarches.append(kokkos_microarch_name)
 
-        for arch in self.amd_gpu_arches:
-            keyval = "amd_gpu_arch=%s" % arch
-            if keyval in spec:
-                spack_microarches.append(arch)
+        if "+rocm" in spec:
+            for amdgpu_target in spec.variants['amdgpu_target'].value:
+                if amdgpu_target != "none":
+                    if amdgpu_target in self.amdgpu_arch_map:
```

Comment:
Could you turn this into a list of `conflict`s so that this is promoted from an installation error to a concretization error? See for instance 

https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/dbcsr/package.py#L57-L59

Also note that 
https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/dbcsr/package.py#L77-L78